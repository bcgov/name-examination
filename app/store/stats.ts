import { getStats } from '~/util/namex-api'
import { getFormattedDateWithTime } from '~/util/date'
import { sortNameChoices } from '~/util'
import { type NameRequest } from '~/types'
import { defineStore } from 'pinia'

export const useStatsStore = defineStore('stats', () => {
  /**
   * Fetches paged statistics data based on the provided timespan and myStats flag.
   *
   * @param {string | null} timespan - The timespan for the statistics data. If null, the default timespan is used.
   * @param {boolean} myStats - A flag indicating whether to fetch statistics related to user's own stats.
   * @return {Promise<any>} A promise that resolves with the fetched statistics data.
   * @throws {Error} Throws an error if the fetching process fails.
   */
  async function getPagedStats (timespan: string | null, myStats: boolean): Promise<any> {
    const params = new URLSearchParams({
      timespan: timespan || '1',
      currentpage: '1',
      perpage: '1000'
    })
    if (myStats) params.append('myStats', 'True')

    try {
      const response = await getStats(params)

      if (response.ok) {
        const jsonData = await response.json() // Parse response JSON

        if (jsonData.numRecords >= 0) {
          const requestsData = jsonData.nameRequests as Array<NameRequest>

          const processedData = requestsData.map((statsData:any) => {
            statsData.names = sortNameChoices(statsData.names)
            if (statsData.comments.length === 1 && statsData.comments[0].timestamp) {
              statsData.comments[0].timestamp = getFormattedDateWithTime(statsData.comments[0].timestamp)
            }
            statsData.lastUpdate = getFormattedDateWithTime(statsData.lastUpdate)
            statsData.names.forEach((name:any) => {
              if (name.decision_text) {
                name.decision_text = name.decision_text.split(/\n\n/)
              }
            })
            return statsData
          })
          return {
            nameRequests: processedData,
            numRecords: jsonData.numRecords
          }
        }
      } else {
        throw new Error('Network response was not ok')
      }
    } catch (error) {
      console.error('There was a problem with your fetch operation:', error)
      throw error // Rethrow the error to handle it elsewhere if needed
    }
  }

  return {
    getPagedStats
  }
})
