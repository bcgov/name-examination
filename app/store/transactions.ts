import {
  ConsentFlag,
  type EntityTypeCode,
  type RequestActionCode,
  type RequestTypeCode,
} from '~/enums/codes'
import { Status } from '~/enums/nr-status'
import {
  type NameChoice,
  type NameRequest,
  type TransactionEntry,
  type Transactions,
} from '~/types'
import { emitter } from '~/util/emitter'
import { getNameRequest, getTransactions } from '~/util/namex-api'
import { toMappedRequestType } from '~/util/request-type'
import requestTypes from '~/data/request_types.json'
import { sortNameChoices } from '~/util'

export const useTransactions = defineStore('transactions', () => {
  const nr = ref<NameRequest>()
  const transactions = ref<Array<TransactionEntry>>([])
  const loadingNr = ref(false)
  const loadingTransactions = ref(false)

  async function loadNr(nrNumber: string) {
    loadingNr.value = true
    const nrResponse = await getNameRequest(nrNumber)
    nr.value = await nrResponse.json()
    loadingNr.value = false
  }

  async function loadTransactions(nrNumber: string) {
    loadingTransactions.value = true
    const transactionsResponse = await getTransactions(nrNumber)
    const transactionsJson = (await transactionsResponse.json()) as Transactions
    transactionsJson.transactions.forEach(t => sortNameChoices(t.names))
    transactions.value = transactionsJson.transactions
    loadingTransactions.value = false
  }

  async function initialize(nrNumber: string) {
    try {
      await loadNr(nrNumber)
      await loadTransactions(nrNumber)
    } catch (e) {
      emitter.emit('error', {
        title: 'Transactions Error',
        message: 'Failed to retrieve transactions data.',
      })
    } finally {
      loadingNr.value = false
      loadingTransactions.value = false
    }
  }

  function getStatusDisplay(status: Status | null, names: Array<NameChoice>) {
    if (!status) return 'N/A'

    let displayState = status as string
    if (status === Status.Conditional) {
      displayState = 'CONDITIONAL APPROVED'
    } else if (status === Status.Consumed && names && names.length > 0) {
      const approvedName = names.find((name) =>
        [Status.Approved, Status.Condition].includes(name.state)
      )
      displayState =
        approvedName?.state === Status.Condition
          ? 'CONDITIONAL APPROVED'
          : 'APPROVED'
      displayState += ` / Used for ${approvedName?.corpNum}`
    }
    return displayState
  }

  function getRequestTypeDisplay(
    requestType: RequestTypeCode,
    action: RequestActionCode
  ) {
    if (!nr.value) return 'N/A'
    const mapped = toMappedRequestType(
      requestType,
      action,
      nr.value.entity_type_cd
    )
    return requestTypes.find((r) => r.value == mapped)?.text || 'N/A'
  }

  function getConsentDisplay(date: string | null, consentFlag: ConsentFlag) {
    return date
      ? 'Required. Received.'
      : consentFlag === ConsentFlag.Required
      ? 'Required. Not yet received.'
      : 'Not required'
  }

  return {
    nr,
    transactions,
    loadingNr,
    loadingTransactions,
    initialize,
    getStatusDisplay,
    getRequestTypeDisplay,
    getConsentDisplay,
  }
})
