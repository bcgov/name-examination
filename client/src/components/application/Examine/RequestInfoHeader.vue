/* eslint-disable */
<template>
  <span>
    <div class="row">

      <!-- details col 1 - priority, comments, etc -->
      <div id='div1' class="col-md-5" >
        <div class="row">
          <div class="col-md-4" style="height: 30px;">
            <div class="priority" v-if="priority">Priority</div>
          </div>
          <div class="col">
            <p v-if="!is_editing" style="font-weight: bold;">{{ requestType_desc }}</p>
            <select v-else v-model="requestType" class="form-control">
              <option v-for="option in requestType_options" v-bind:value="option.value"
                      v-bind:key="option.value">
                {{ option.text }}
              </option>
            </select>
          </div>
        </div>

        <div class="row">
          <div class="col-md-4" >
            <div class="nrNum" v-bind:class="{ REDnrNum: priority}">{{ nrNumber }}</div>
            <p v-if="!is_editing"
               style="font-weight: bold; text-align: center;">{{ jurisdiction_desc }}</p>
          </div>
          <div class="col">
            <nwpta jurisdiction="AB" />
          </div>
          <div class="col">
            <nwpta jurisdiction="SK" />
          </div>
        </div>
        <div class="row">
          <div class="col-md-4" >
            <!-- spacer -->
          </div>
          <div class="col add-top-padding" >
            <h3 v-if="is_editing">Jurisdiction</h3>
            <select v-if="is_editing" v-model="jurisdiction" class="form-control">
              <option v-for="option in jurisdiction_options" v-bind:value="option.value"
                      v-bind:key="option.value">
                {{ option.text }}
              </option>
            </select>
          </div>
        </div>

        <div class="row">
          <div class="col add-top-padding">
            <div v-if="show_extended_header">
              <div>
                <h3>INTERNAL COMMENTS</h3>
                <div class="comment" v-for="comment in internalComments"
                     v-bind:key="comment.timestamp">
                  <p>
                    <span class="comment-examiner">{{ comment.examiner }}</span>
                    -
                    <span class="comment-timestamp">{{ comment.timestamp }}</span>
                  </p>
                  <p class="comment-text">{{ comment.comment }}</p>

                </div>
              </div>
              <div v-if="is_editing">
                <h3>add comment</h3>
                <textarea v-model="newComment" class="form-control" rows="5"></textarea>
              </div>
            </div>
          </div>
        </div>
      </div>

      <!-- details col 2 - nature of business -->
      <div id='div2' class="col-md-4">
        <div class="row">
          <div class="col">
            <h3>SUBMITTED DATE</h3>
            {{submittedDate}}

            <h3 v-if="expiryDate !== null">EXPIRY DATE</h3>
            <span v-if="expiryDate !== null">{{expiryDate}}</span>

            <h3 v-if="consumptionDate !== null">CONSUMPTION DATE</h3>
            <span v-if="consumptionDate !== null">{{consumptionDate}}</span>

            <h3 v-if="submitCount > 0">SUBMIT COUNT: {{submitCount}}</h3>

            <h3>NATURE OF BUSINESS</h3>
            <div v-if="show_extended_header">
              <textarea v-if="is_editing" v-model="natureOfBusiness" class="form-control" rows="10">
                </textarea>
              <p v-else style="white-space: pre-line;">{{ natureOfBusiness }}</p>
            </div>
            <p v-else style="white-space: pre-line;">{{ natureOfBusinessTruncated }}</p>

          </div>

        </div>
        <div class="row" v-if="is_editing">
          <div class="col add-top-padding">
            <h3>Name Choices</h3>

            <table style="width: 100%;">
              <tr>
                <td>1.</td>
                <td><input v-model="compName1.name" class="form-control" /></td>
              </tr>
              <tr>
                <td>2.</td>
                <td><input v-model="compName2.name" class="form-control" /></td>
              </tr>
              <tr>
                <td>3.</td>
                <td><input v-model="compName3.name" class="form-control" /></td>
              </tr>
            </table>
          </div>
        </div>
      </div>

      <!-- details col 3 - additional info, client details -->
      <div id='div3' class="col-md-3">
        <div class="row">
          <div class="col">
            <div class="row">
              <div class="col">
                <h3>ADDITIONAL INFORMATION</h3>
                <p v-if="!is_editing" style="white-space: pre-line;">{{ additionalInfo }}</p>
                <textarea v-else v-model="additionalInfo" class="form-control"></textarea>
              </div>
            </div>
            <div class="row">
              <div class="col add-top-padding">
                <clientinfoview />
              </div>
            </div>
          </div>
        </div>
      </div>

    </div>

    <!-- row 2 - buttons -->
    <div class="row">
      <div id='header-button-container' class="col-md-12">

        <button v-if="!is_editing" class="f1 btn btn-sm btn-outline-secondary"
                @click="toggleDetails">Show Details</button>

        <button v-if="!is_editing" class="btn btn-sm btn-secondary"
                style="float: left;" @click="edit">Edit</button>

        <button v-if="is_editing" class="btn btn-sm btn-success"
                style="float: left;" @click="save">Save</button>
        <button v-if="is_editing" class="btn btn-sm btn-secondary"
                style="float: left;" @click="cancelSave">Cancel</button>

      </div>
    </div>

  </span>
</template>

<script>
/* eslint-disable */

// ClientInfoHeader - editable component
import clientinfoview from '@/components/application/Examine/client/ClientInfoHeader.vue';
import nwpta from '@/components/application/Examine/nwpta/nwpta.vue';

export default {
    name: 'RequestInfoHeader',
    data: function () {
      return {
        newComment: null,
      }
    },
    computed: {
      jurisdiction_options() {
        return this.$store.getters.listJurisdictions;
      },
      requestType_options() {
        return this.$store.getters.listRequestTypes;
      },
      is_editing() {
        return  this.$store.getters.is_editing;
      },
      show_extended_header() {
        return this.is_editing || this.$store.state.is_header_shown;
      },
      nrNumber() {
        return  this.$store.getters.nrNumber;
      },
      compName1: {
        get: function() {
          return this.$store.getters.compName1;
        },
        set: function(value) {
          this.$store.commit('compName1', value);
        }
      },
      compName2: {
        get: function() {
          return this.$store.getters.compName2;
        },
        set: function(value) {
          this.$store.commit('compName2', value);
        }
      },
      compName3: {
        get: function() {
          return this.$store.getters.compName3;
        },
        set: function(value) {
          this.$store.commit('compName3', value);
        }
      },
      requestType: {
        get: function() {
          return this.$store.getters.requestType;
        },
        set: function(value) {
          this.$store.commit('requestType', value);
        }
      },
      requestType_desc() {
        try {
          return getDescFromList(this.requestType_options, this.requestType);
        } catch (err) {
          return 'ERROR!!';
        }
      },
      jurisdiction: {
        get: function() {
          return this.$store.getters.jurisdiction;
        },
        set: function(value) {
          this.$store.commit('jurisdiction', value);
        }
      },
      jurisdiction_desc() {
        try {
          return getDescFromList(this.jurisdiction_options, this.jurisdiction);
        } catch (err) {
          return '';
        }
      },
      natureOfBusiness: {
        get: function() {
          return  this.$store.getters.natureOfBusiness;
        },
        set: function(value) {
          this.$store.commit('natureOfBusiness', value);
        }
      },
      natureOfBusinessTruncated() {
        try {
          if (this.natureOfBusiness.length > 200) return this.natureOfBusiness.substr(0, 200) + '...';
          else return this.natureOfBusiness;
        } catch (err) {
          return this.natureOfBusiness;
        }
        if (this.natureOfBusiness.length > 200) return this.natureOfBusiness.substr(0, 200) + '...';
        else return this.natureOfBusiness;
      },
      nr_status: {
        get: function() {
          return this.$store.getters.nr_status;
        },
        set: function(value) {
          this.$store.commit('nr_status', value);
        }
      },
      examiner() {
        return this.$store.getters.examiner;
      },
      priority() {
        return this.$store.getters.priority;
      },
      resubmissionYN: {
        get: function() {
          return this.$store.getters.resubmissionYN;
        },
        set: function(value) {
          this.$store.commit('resubmissionYN', value);
        }
      },
      linkedNR: {
        get: function() {
          return this.$store.getters.linkedNR;
        },
        set: function(value) {
          this.$store.commit('linkedNR', value);
        }
      },
      reservationCount: {
        get: function() {
          return this.$store.getters.reservationCount;
        },
        set: function(value) {
          this.$store.commit('reservationCount', value);
        }
      },
      details: {
        get: function() {
          return this.$store.getters.details;
        },
        set: function(value) {
          this.$store.commit('details', value);
        }
      },
      additionalInfo: {
        get: function() {
          return this.$store.getters.additionalInfo;
        },
        set: function(value) {
          this.$store.commit('additionalInfo', value);
        }
      },
      internalComments: {
        get: function() {
          return this.$store.getters.internalComments;
        },
        set: function(value) {
          this.$store.commit('internalComments', value);
        }
      },
      expiryDate() {
        return this.$store.getters.expiryDate;
      },
      consumptionDate() {
        return this.findConsumptionDate();
      },
      submittedDate() {
        return this.$store.getters.submittedDate;
      },
      submitCount() {
        return this.$store.getters.submitCount;
      },
    },
    components: {
      clientinfoview,
      nwpta,
    },
    methods: {
      toggleDetails() {
        if (this.$store.state.is_header_shown) this.$store.state.is_header_shown = false;
        else this.$store.state.is_header_shown = true;
      },
      edit() {
        this.$store.state.is_editing = true;
      },
      save() {

        // save new comment
        this.addNewComment();

        this.$store.dispatch('updateRequest');
        this.$store.state.is_editing = false;

        // get updated data fresh from server
        this.$store.dispatch('getpostgrescompInfo',this.nrNumber)

        // show full header after editing so user can see everything they changed
        this.$store.state.is_header_shown = true;

      },
      cancelSave() {
        this.$store.dispatch('getpostgrescompInfo',this.nrNumber)
        this.$store.state.is_editing = false;
      },
      addNewComment() {

        // create new comment object with just text, and add it to list of comments in data structure
        var newCommentData = {
          comment: this.newComment,
          examiner: this.$store.state.examiner
        };
        this.internalComments = this.internalComments.concat(newCommentData);

        // clear newComment field for next comment added in this session
        this.newComment = null;
      },
      findConsumptionDate() {
        if (this.compName1.consumptionDate != null) {
          return this.compName1.consumptionDate;
        }
        if (this.compName2.consumptionDate != null) {
          return this.compName2.consumptionDate;
        }
        return this.compName3.consumptionDate;
      }
    },
    watch: {
      nrNumber: function (val) {
        console.log('RequestInfoHeader.nrNumber watcher fired:' )
        this.$store.dispatch('getpostgrescompInfo',this.nrNumber)
        if( this.$store.getters.currentChoice == null || this.$store.getters.currentChoice == 1 ){
          console.log('RequestInfoHeader.watch runRecipe:' + val)
          this.$store.dispatch('runRecipe')
        } else {
          console.log('RequestInfoHeader.watch set currentChoice to 1')
          this.$store.commit('currentChoice',1)
        }
      }
    }
}
</script>

<style scoped>

   .nrNum {
     font-size: 1.5em;
     font-weight: bold;
     text-align: center;
     margin-top: 10px;
   }
  .REDnrNum {
    color: #ff0000;
  }
  .priority {
    padding: 5px;
    text-align: center;
    background-color: #cc0000;
    color: white;
  }

  #header-button-container button {
    margin: 5px 5px 5px 0;
  }
  #header-button-container .f1 {
    float: right;
    margin-right: 0;
  }

  .comment {
    padding: 10px 0px;
    border-bottom: 1px solid #ccc;
  }
  .comment:last-child {
    border-bottom: none;
  }

  .comment-text {
    white-space: pre-line;
  }

  .comment-timestamp, .comment-examiner {
    font-style: italic;
  }



 </style>
