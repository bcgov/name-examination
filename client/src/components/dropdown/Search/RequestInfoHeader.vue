/* eslint-disable */
<template>
  <span>
    <div class="row">

      <!-- details col 1 - priority, comments, etc -->
      <div id='div1' class="col-md-5" >
        <div class="row">
          <div class="col-md-3" >
            <span v-if="priority" id="lblPriority" class="ISPRIORITY" >Priority</span>
            <span v-else></span>
          </div>
          <div class="col-md-9">
            <p v-if="!is_editing">{{ requestType }}</p>
            <textarea v-else
                      id="requestType1"
                      v-model="requestType"
                      class="txtArea form-control">
            </textarea>
          </div>
        </div>

        <div class="row">
          <div class="col-md-3" >
            <p class="nr-number" v-bind:class="{ REDnrNum: priority }">{{ nrNumber }}</p>
          </div>
          <div class="col" >
            <p v-if="!is_editing">{{ jurisdiction }}</p>
            <input v-else v-model="jurisdiction" class="form-control" />
          </div>
          <div class="col">
            <p v-if="!is_editing">{{ nuans }}</p>
            <input v-else v-model="nuans" class="form-control" />
          </div>
          <div class="col">
            <p v-if="!is_editing">{{ sk_name }}</p>
            <input v-else v-model="sk_name" class="form-control" />
          </div>
        </div>

        <div class="row">
          <div class="col add-top-padding">
            <div id="internalCommentsDisplay" v-if="!is_editing" style="display: none;">
              <h3>INTERNAL COMMENTS</h3>
              <p>{{ internalComments }}</p>
            </div>
            <div v-else>
              <h3>INTERNAL COMMENTS</h3>
              <textarea v-model="internalComments" class="form-control"></textarea>
            </div>

          </div>
        </div>
      </div>

      <!-- details col 2 - nature of business -->
      <div id='div2' class="col-md-4">
        <h3>NATURE OF BUSINESS</h3>
        <p id="natureOfBusinessDisplay" v-if="!is_editing">{{ natureOfBusiness }}</p>
        <textarea v-else v-model="natureOfBusiness" class="form-control"></textarea>

      </div>

      <!-- details col 3 - additional info, client details -->
      <div id='div3' class="col-md-3">
        <div class="row">
          <div class="col">
            <div class="row">
              <div class="col">
                <h3>ADDITIONAL INFORMATION</h3>
                <p v-if="!is_editing">{{ addInfo }}</p>
                <textarea v-else v-model="addInfo" class="form-control"></textarea>
              </div>
            </div>
            <div class="row">
              <div class="col add-top-padding">
                <clientinfoview />
              </div>
            </div>
          </div>
        </div>
      </div>

    </div>

    <!-- row 2 - buttons -->
    <div class="row">
      <div id='div4' class="col-md-12">
        <span class="f1" @click="toggleDetails">F1</span>
      </div>
      <button v-if="!is_editing" class="btn btn-default" @click="edit">Edit</button>
      <button v-else class="btn btn-default" @click="save">Save</button>
    </div>

  </span>
</template>

<script>
/* eslint-disable */

import clientinfoview from '@/components/dropdown/Search/client/ClientInfo.vue';

export default {
    name: 'RequestInfoHeader',
    computed: {
      is_editing() {
        return  this.$store.getters.is_editing;
      },
      nrNumber() {
        return  this.$store.getters.nrNumber;
      },
      requestType() {
        return  this.$store.getters.requestType;
      },
      firstName() {
        return  this.$store.getters.firstName;
      },
      lastName() {
        return  this.$store.getters.lastName;
      },
      address() {
        return  this.$store.getters.address;
      },
      contactName() {
        return  this.$store.getters.contactName;
      },
      phone() {
        return  this.$store.getters.phone;
      },
      conEmail() {
        return  this.$store.getters.conEmail;
      },
      fax() {
        return  this.$store.getters.fax;
      },
      jurisdiction() {
        return  this.$store.getters.jurisdiction;
      },
      natureOfBusiness() {
        return  this.$store.getters.natureOfBusiness;
      },
      nuans() {
        return  this.$store.getters.nuans;
      },
      sk_name() {
        return  this.$store.getters.sk_name;
      },
      nr_status() {
        return  this.$store.getters.nr_status;
      },
      examiner() {
        return  this.$store.getters.examiner;
      },
      priority() {
        return  this.$store.getters.firstName;
      },
      resubmissionYN() {
        return  this.$store.getters.resubmissionYN;
      },
      linkedNR() {
        return  this.$store.getters.linkedNR;
      },
      reservationCount() {
        return  this.$store.getters.reservationCount;
      },
      expiryDate() {
        return  this.$store.getters.expiryDate;
      },
      details() {
        return this.$store.getters.details;
      },
      internalComments() {
        // TODO get actual data from $store
        return 'Lorem ipsum dolor sit amet, at vix enim nominavi, ea dico case euismod mea, est cu ubique iuvaret. Vis ea rebum verear, salutandi posidonium signiferumque ut qui. Et inermis iudicabit sententiae his, tation facilisis eam id. Adhuc phaedrum inimicus cu sit.\n' +
          '\n' +
          'Ullum melius delicatissimi te sit, ut nullam lucilius rationibus has. Mei sonet moderatius at. Pro legere mentitum omittantur te, natum decore cu pro. Ei brute labores mea, ea assum omittantur nec. Ea per mnesarchum elaboraret necessitatibus, graece nemore id pri. Ut saepe salutatus intellegam sit, magna vivendo nam in, minimum lucilius qui ne. Mei ei quis verear vulputate, cu vis adipisci ullamcorper.\n' +
          '\n' +
          'Sed alterum convenire signiferumque an, quando omittam repudiandae vix id, nec oratio equidem ei. Ea nec tractatos periculis. Dolore pertinacia id mel, pro patrioque forensibus ea. Aperiri virtute sea et.\n' +
          '\n' +
          'Oportere electram in eos, volumus pericula mei cu. Ius suscipit vivendum scripserit in, ne graecis detraxit patrioque nec. Nec laudem aliquid et, hinc vitae nullam est eu. No errem oblique vix, sed no augue novum dicant, ne purto modus phaedrum sed. Inimicus accusamus mei cu. Qui an altera saperet, fierent percipitur ius no, ex mea ignota quaestio.\n' +
          '\n' +
          'Ea dictas omittam nam, in his fugit mediocrem, vim no invidunt indoctum. Veri audire ne eos. Vim everti fierent an, ad usu omnes adolescens honestatis, tempor detraxit at quo. Duo in zril soluta, vim ne ferri periculis theophrastus. Idque utamur quaeque in eam, te eam deleniti periculis.\n';
      }
    },
    components: {
      clientinfoview
    },
    mounted(){
      this.setInterface();
    },
    methods: {
      setInterface(){
        this.setNRNum()
        this.setPriority()
        this.setComp
        this.setNuans()
        this.setSK()
        this.setNOB()
        this.setAddInfo()
      },
      setNRNum() {
      },
      setPriority() {
      },
      setAddInfo() {
      },
      setComp() {
      },
      setNuans() {
      },
      setSK() {
      },
      setNOB() {
      },
      toggleDetails() {
        // KATIE - IN PROGRESS

        // show internal comments
        document.getElementById("internalCommentsDisplay").style.display = '';

        // show complete nature of business, untruncated

        // show client data
        document.getElementsByClassName('RequestInfoHeader')[0].getElementsByClassName('ClientInfo')[0].style.display = 'inline';

      },
      setBorder(id) {
        var tb = document.getElementById(id);
        tb.borderWidth = "1";
      },
      edit() {
        this.$store.state.is_editing = true;
      },
      save() {
        this.$store.state.is_editing = false;
      }
    },
    watch: {
      nrNumber: function (val) {
        this.$store.dispatch('getpostgrescompInfo',this.nrNumber)
        this.setInterface();
      }
    }
}
</script>

<style scoped>
   .nrNum {
     width: 100%;
     background-color: #f1f1f1;
     border: 0px #fff;
     color: #ff0000;
     font-size: 1.9em;
   }
  .REDnrNum {
    width: 100%;
    background-color: #f1f1f1;
    color: #ff0000;
    border: 0px #fff;
  }
  .ISPRIORITY {
    padding-left: 28%;
    padding-right: 28%;
    padding-bottom: 6%;
    padding-top: 6%;
    border: 2px solid #777;
    background-color: #cc0000;
    font-size: 1.25em;
    color: #ffffff;
  }
   .f1 {
     border: 1px solid #000000;
     padding: 2px;
   }
   /* initially hide client info ONLY WITHIN HEADER (not everywhere the component is used */
   .RequestInfoHeader .ClientInfo {
     display: none;
   }
 </style>
