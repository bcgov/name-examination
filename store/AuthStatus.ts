import { defineStore } from 'pinia';

export const useAuthStore = defineStore('auth', {
  state: () => ({
    auth: false,
  }),
  actions: {
    toggleAuth() {
      this.auth = !this.auth;
    },
  },
});
