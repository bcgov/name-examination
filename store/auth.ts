import ConfigHelper from '../util/config-helper'
import KeycloakServices from '../public/keycloak/keycloak'
import { SessionStorageKeys } from '../util/constants'
import { defineStore } from 'pinia'

interface AuthState {
  token: string;
  idToken: string;
  refreshToken: string;
  kcGuid: string;
  loginSource: string;
}

export const useAuthStore = defineStore({
  id: 'auth',
  state: (): AuthState => ({
    token: '',
    idToken: '',
    refreshToken: '',
    kcGuid: '',
    loginSource: ''
  }),
  getters: {
    isAuthenticated (state): boolean {
      return !!state.token
    },
    keycloakGuid (state): string {
      return state.kcGuid || KeycloakServices.getUserInfo().keycloakGuid
    },
    currentLoginSource (state): string {
      return state.loginSource || KeycloakServices.getUserInfo().loginSource
    }
  },
  actions: {
    setKCToken (token: string): void {
      this.token = token
      ConfigHelper.addToSession(SessionStorageKeys.KeyCloakToken, token)
    },
    setIDToken (idToken: string): void {
      this.idToken = idToken
      ConfigHelper.addToSession(SessionStorageKeys.KeyCloakIdToken, idToken)
    },
    setRefreshToken (refreshToken: string): void {
      this.refreshToken = refreshToken
      ConfigHelper.addToSession(SessionStorageKeys.KeyCloakRefreshToken, refreshToken)
    },
    setKCGuid (kcGuid: string): void {
      this.kcGuid = kcGuid
    },
    setLoginSource (loginSource: string): void {
      this.loginSource = loginSource
    },
    clearSession (): void {
      // Reset all state properties related to the user's session
      this.token = ''
      this.idToken = ''
      this.refreshToken = ''
      this.kcGuid = ''
      this.loginSource = ''

      // Clear the session storage values
      ConfigHelper.removeFromSession(SessionStorageKeys.KeyCloakToken)
      ConfigHelper.removeFromSession(SessionStorageKeys.KeyCloakIdToken)
      ConfigHelper.removeFromSession(SessionStorageKeys.KeyCloakRefreshToken)
    },
    syncWithSessionStorage (): void {
      const token = ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakToken) || ''
      const idToken = ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakIdToken) || ''
      const refreshToken = ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakRefreshToken) || ''

      this.token = token
      this.idToken = idToken
      this.refreshToken = refreshToken
    }
  }
})

if (import.meta.hot) {
  import.meta.hot.accept(acceptHMRUpdate(useAuthStore, import.meta.hot))
}
