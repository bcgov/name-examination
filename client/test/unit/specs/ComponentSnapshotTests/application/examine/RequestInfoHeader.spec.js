/* eslint-disable */
import Vue from 'vue';
Vue.use(require('vue-shortkey'))
import RequestInfoHeader from '@/components/application/Examine/RequestInfoHeader';
import store from '@/store'

describe('RequestInfoHeader.vue', () => {

   let component, vm;
    beforeEach(() => {
        component = Vue.extend(RequestInfoHeader);
        vm = new component({store: store});
    });

   it("renders an RequestInfoHeader component", () => {
     expect(vm.$el).toMatchSnapshot();
   })

  // Add other tests specific to this component and not its sub-components
});
