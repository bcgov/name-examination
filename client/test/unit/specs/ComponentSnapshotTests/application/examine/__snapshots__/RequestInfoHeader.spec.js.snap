// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestInfoHeader.vue renders an RequestInfoHeader component 1`] = `undefined`;
